tests/test.sh

```diff
@@ -0,0 +1,163 @@
+    export TEST_NUM_CPU_SERVER=${TEST_NUM_CPU_SERVER:-1}
+    export DMLC_NUM_GPU_DEV=${DMLC_NUM_GPU_DEV:-0}
+    export DMLC_NUM_CPU_DEV=${DMLC_NUM_CPU_DEV:-1}
+elif [ "$dev_type" == "cpu2gpu" ]; then
+    export TEST_NUM_GPU_WORKER=${TEST_NUM_GPU_WORKER:-0}
+    export TEST_NUM_GPU_SERVER=${TEST_NUM_GPU_SERVER:=1}
+    export TEST_NUM_CPU_WORKER=${TEST_NUM_CPU_WORKER:-1}
+    export TEST_NUM_CPU_SERVER=${TEST_NUM_CPU_SERVER:-0}
+    export ENABLE_RECV_BUFFER=${ENABLE_RECV_BUFFER:=1}
+    export DMLC_NUM_GPU_DEV=${DMLC_NUM_GPU_DEV:-1}
+    export DMLC_NUM_CPU_DEV=${DMLC_NUM_CPU_DEV:-0}
+elif [ "$dev_type" == "gpu2gpu" ]; then
+    export TEST_NUM_GPU_WORKER=${TEST_NUM_GPU_WORKER:-1}
+    export TEST_NUM_GPU_SERVER=${TEST_NUM_GPU_SERVER:=1}
+    export TEST_NUM_CPU_WORKER=${TEST_NUM_CPU_WORKER:-0}
+    export TEST_NUM_CPU_SERVER=${TEST_NUM_CPU_SERVER:-0}
+    export ENABLE_RECV_BUFFER=${ENABLE_RECV_BUFFER:=1}
+    export DMLC_NUM_GPU_DEV=${DMLC_NUM_GPU_DEV:-1}
+    export DMLC_NUM_CPU_DEV=${DMLC_NUM_CPU_DEV:-0}
+elif [ "$dev_type" == "gpu2cpu" ]; then
+    export TEST_NUM_GPU_WORKER=${TEST_NUM_GPU_WORKER:-1}
+    export TEST_NUM_GPU_SERVER=${TEST_NUM_GPU_SERVER:=0}
+    export TEST_NUM_CPU_WORKER=${TEST_NUM_CPU_WORKER:-0}
+    export TEST_NUM_CPU_SERVER=${TEST_NUM_CPU_SERVER:-1}
+    export DMLC_NUM_GPU_DEV=${DMLC_NUM_GPU_DEV:-1}
+    export DMLC_NUM_CPU_DEV=${DMLC_NUM_CPU_DEV:-0}
+else
+    echo "Unexpected device type. Please choose among: (cpu2cpu|cpu2gpu|gpu2gpu|gpu2cpu)"
+    exit -1
+fi
+
+export ARGS=${ARGS:-$bytes_per_msg 100000 $op_code}
+export NUM_KEY_PER_SERVER=$msg_count
+
+export LD_LIBRARY_PATH=$CUDA_HOME/lib64:$UCX_HOME/lib:$LD_LIBRARY_PATH
+# the number of workers
+export DMLC_NUM_WORKER=${DMLC_NUM_WORKER:-1}
+# the number of servers
+export DMLC_NUM_SERVER=${DMLC_NUM_SERVER:=$DMLC_NUM_WORKER}
+export DMLC_PS_ROOT_URI=${NODE_ONE_IP}
+export DMLC_PS_ROOT_PORT=${DMLC_PS_ROOT_PORT:-12278} # scheduler's port (can random choose)
+
+# ==============================
+# ======= RDMAVan options ======
+# ==============================
+# enable rdma
+export DMLC_ENABLE_RDMA=${DMLC_ENABLE_RDMA:-1}
+# NIC for RDMAVan
+export DMLC_INTERFACE=eth2
+
+
+# =======================================
+# ===== preset default UCX vars =========
+# =======================================
+export UCX_USE_MT_MUTEX=y
+export UCX_SOCKADDR_CM_ENABLE=y
+export UCX_RNDV_THRESH=8k
+export UCX_IB_TRAFFIC_CLASS=236
+export UCX_TLS=rc_x,ib,sm,tcp,cuda
+export UCX_LOG_LEVEL=info
+# export UCX_IB_NUM_PATHS=${UCX_IB_NUM_PATHS:=2}
+# export UCX_MEMTYPE_CACHE=n
+# export UCX_RNDV_SCHEME=put_zcopy
+# export UCX_IB_GPU_DIRECT_RDMA=no
+# export UCX_MAX_RNDV_RAILS=2
+
+# ==============================
+# ======== UCXVan options ======
+# ==============================
+# enable ucx
+export DMLC_ENABLE_UCX=${DMLC_ENABLE_UCX:-1}
+export UCX_NET_DEVICES=${UCX_NET_DEVICES:=eth0,eth2,mlx5_2:1}
+export BYTEPS_UCX_SHORT_THRESH=0
```

Comment:
do you see any bugs when this var is not set?

Reply:
updated to 4096